### PR TITLE
[C-1974] Add batch remove downloads

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Adds batch remove downloads flow
Interested to see what yall think regarding the architecture. In this case, remove-downloads-button dispatches an action, and that action runs a saga which then batch updates the redux store, which triggers more sagas to save to disk. With this architecture we have a potentially simpler data flow.

TODO: need to fit in the queue cancel. would like to talk about that a bit to make sure we are cancelling the correct jobs

### CODE REVIEW

File: packages/mobile/src/store/offline-downloads/sagas/removeAllDownloadedFavoritesSaga.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+function* removeAllFavoritesTracks() {
+  const offlineTracks = yield* select(getOfflineTracks)
+  const offlineTrackList = Object.keys(offlineTracks).map(
+    (offlineTrackId) => offlineTracks[offlineTrackId]
+  )
+
+  const tracksToRemove: ID[] = []
+  const reasonsToUpdate: TrackReasonsToUpdate[] = []
+
+  // TODO: double check if "favorites" collection is in here
```
Comment (sliptype) [inline packages/mobile/src/store/offline-downloads/sagas/removeAllDownloadedFavoritesSaga.ts:49]: Is this done?

File: packages/mobile/src/store/offline-downloads/sagas/removeAllDownloadedFavoritesSaga.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  const offlineFavoritedCollections = yield* select(
+    getOfflineFavoritedCollections
+  )
+  const favoritedCollectionIds = new Set([
+    'favorites',
+    ...Object.keys(offlineFavoritedCollections)
+  ])
+
+  offlineTrackList.forEach((offlineTrack) => {
```
Comment (sliptype) [inline packages/mobile/src/store/offline-downloads/sagas/removeAllDownloadedFavoritesSaga.ts:58]: Apparently `for ... of` is more performant than `.forEach`

**Review by sliptype — APPROVED** [review]
I like this architecture a lot! Also having the sagas in separate named files is so nice

File: packages/mobile/src/store/offline-downloads/sagas/removeAllDownloadedFavoritesSaga.ts (hunk 1 above)
Comment (dylanjeffers) [reply packages/mobile/src/store/offline-downloads/sagas/removeAllDownloadedFavoritesSaga.ts:49]: yes thanks

File: packages/mobile/src/store/offline-downloads/sagas/removeAllDownloadedFavoritesSaga.ts (hunk 2 above)
Comment (dylanjeffers) [reply packages/mobile/src/store/offline-downloads/sagas/removeAllDownloadedFavoritesSaga.ts:58]: down with that

**Pull request merged**